src/pages/PlanPage/subRoutes/UpgradePlanPage/UpgradeForm/ProPlanControls/ProPlanControls.spec.jsx

```diff
@@ -0,0 +1,915 @@
+        </MemoryRouter>
+      </QueryClientProvider>
+    )
+
+describe('ProPlanControls', () => {
+  function setup(
+    {
+      planValue = Plans.USERS_BASIC,
+      successfulPatchRequest = true,
+      errorDetails = undefined,
+      trialStatus = undefined,
+    } = {
+      planValue: Plans.USERS_BASIC,
+      successfulPatchRequest: true,
+      errorDetails: undefined,
+      trialStatus: undefined,
+    }
+  ) {
+    const addNotification = jest.fn()
+    const user = userEvent.setup()
+    const patchRequest = jest.fn()
+
+    useAddNotification.mockReturnValue(addNotification)
+
+    server.use(
+      rest.get(`/internal/gh/codecov/account-details/`, (req, res, ctx) => {
+        if (planValue === Plans.USERS_BASIC) {
+          return res(ctx.status(200), ctx.json(mockAccountDetailsBasic))
+        } else if (planValue === Plans.USERS_PR_INAPPM) {
+          return res(ctx.status(200), ctx.json(mockAccountDetailsProMonthly))
+        } else if (planValue === Plans.USERS_PR_INAPPY) {
+          return res(ctx.status(200), ctx.json(mockAccountDetailsProYearly))
+        } else if (planValue === Plans.USERS_TRIAL) {
+          return res(ctx.status(200), ctx.json(mockAccountDetailsTrial))
+        }
+      }),
+      rest.patch(
+        '/internal/gh/codecov/account-details/',
+        async (req, res, ctx) => {
+          if (!successfulPatchRequest) {
+            if (errorDetails) {
+              return res(ctx.status(500), ctx.json({ detail: errorDetails }))
+            }
+            return res(ctx.status(500), ctx.json({ success: false }))
+          }
+          const body = await req.json()
+
+          patchRequest(body)
+
+          return res(ctx.status(200), ctx.json({ success: false }))
+        }
+      ),
+      graphql.query('GetAvailablePlans', (req, res, ctx) => {
+        return res(
+          ctx.status(200),
+          ctx.data({
+            owner: { availablePlans: allPlans },
+          })
+        )
+      }),
+      graphql.query('GetPlanData', (req, res, ctx) => {
+        return res(
+          ctx.status(200),
+          ctx.data({
+            owner: {
+              plan: { ...mockPlanDataResponse, trialStatus },
+            },
+          })
+        )
+      })
+    )
+
+    return { addNotification, user, patchRequest }
+  }
+
+  describe('when rendered', () => {
+    describe('when the user has a basic plan', () => {
+      it('renders organization title', async () => {
+        setup({ planValue: Plans.USERS_BASIC })
+        render(<ProPlanControls />, { wrapper: wrapper() })
+
+        const organizationTitle = await screen.findByText(/Organization/)
+        expect(organizationTitle).toBeInTheDocument()
+      })
+
+      it('renders the owner', async () => {
+        setup({ planValue: Plans.USERS_BASIC })
+        render(<ProPlanControls />, { wrapper: wrapper() })
+
+        const ownerTitle = await screen.findByText(/codecov/)
+        expect(ownerTitle).toBeInTheDocument()
+      })
```

Comment:
nitpick: I would combine these into a single test, as in you're testing the section, just to reduce some lines from the file 😅

Reply:
Suree